tests/test_manifest.py

```diff
@@ -763,39 +763,77 @@ def test_create_manifests(
             assert all_results == expected_result
 
     @pytest.mark.parametrize(
-        "component,datasetId",
-        [("Biospecimen", "syn61260107"), ("BulkRNA-seqAssay", "syn61374924")],
+        "component,datasetId,expected_file_based,expected_rows,expected_files",
+        [
+            ("Biospecimen", "syn61260107", False, 4, None),
+            (
+                "BulkRNA-seqAssay",
+                "syn61374924",
+                True,
+                4,
+                pd.Series(
+                    [
+                        "schematic - main/BulkRNASeq and files/txt1.txt",
+                        "schematic - main/BulkRNASeq and files/txt2.txt",
+                        "schematic - main/BulkRNASeq and files/txt4.txt",
+                        "schematic - main/BulkRNASeq and files/txt3.txt",
+                    ],
+                    name="Filename",
+                ),
+            ),
+        ],
         ids=["Record based", "File based"],
     )
-    def test_get_manifest_with_files(self, helpers, component, datasetId):
+    def test_get_manifest_with_files(
+        self,
+        helpers,
+        component,
+        datasetId,
+        expected_file_based,
+        expected_rows,
+        expected_files,
+    ):
         """
-        Test to ensure that when generating a record based manifset that has files in the dataset that the files are not added to the manifest as well
+        Test to ensure that
+            when generating a record based manifset that has files in the dataset that the files are not added to the manifest as well
+            when generating a file based manifest from a dataset thathas had files added that the files are added correctly
         """
+        # GIVEN the example data model
```

Comment:
Fantastic usage of what's learned from a PR.  Love to see it!

Reply:
Thanks @thomasyu888 !